florist/api/monitoring/metrics.py

```diff
@@ -64,3 +65,56 @@ def dump(self) -> None:
         encoded_metrics = json.dumps(self.metrics, cls=DateTimeEncoder)
         log(DEBUG, f"Dumping metrics to redis at key '{self.run_id}': {encoded_metrics}")
         self.redis_connection.set(self.run_id, encoded_metrics)
+
+
+def wait_for_metric(
+    uuid: str,
+    metric: str,
+    redis_host: str,
+    redis_port: str,
+    logger: Logger,
+    max_retries: int = 20,
+    seconds_to_sleep_between_retries: int = 1,
+) -> None:
+    """
+    Check metrics on Redis under the given UUID and wait until it appears.
+
+    If the metrics are not there yet, it will retry up to max_retries times,
+    sleeping and amount of seconds_to_sleep_between_retries between them.
+
+    :param uuid: (str) The UUID to pull the metrics from Redis.
+    :param metric: (str) The metric to look for.
+    :param redis_host: (str) The hostname of the Redis instance the metrics are being reported to.
+    :param redis_port: (str) The port of the Redis instance the metrics are being reported to.
+    :param logger: (logging.Logger) A logger instance to write logs to.
+    :param max_retries: (int) The maximum number of retries. Optional, default is 20.
+    :param seconds_to_sleep_between_retries: (int) The amount of seconds to sleep between retries.
+        Optional, default is 1.
+    :raises Exception: If it retries MAX_RETRIES times and the right metrics have not been found.
```

Comment:
nit: not sure if you intended on capitalizing MAX_RETRIES

Reply:
Yes, it used to be a constant then I changed it to a param and forgot to update it here :)